gobblin-docs/sinks/Http.md

```diff
@@ -0,0 +1,105 @@
+| `body` | Optional, the request body in string or json string format | ```"{\"email\": \"httpwrite@test.com\"}"``` |
+
+Given an url template, ```http://www.test.com/profiles/${memberId}```, from job configuration, the resolved 
+example request url with `keys` and `queryParams` information will be ```http://www.test.com/profiles/123?action=update```.
+
+## `AsyncRequestBuilder`
+An `AsyncRequestBuilder` builds an `AsyncRequest` from a collection of `HttpOperation` records. It could build one
+request per record or batch multiple records into a single request. A builder is also responsible for
+putting the `headers` and setting the `body` to the request.
+
+## `HttpClient`
+A `HttpClient` sends a request and returns a response. If necessary, it should setup the connection to the server, for
+example, sending an authorization request to get access token.
+
+## `ResponseHandler`
+A `ResponseHandler` handles a response of a request. It returns a `ResponseStatus` object to the framework, which
+would resend the request if it's a `SERVER_ERROR`.
+
+# Build an asynchronous writer
+`AsyncHttpWriterBuilder` is the base builder to build an asynchronous http writer. A specific writer can be created by 
+providing the 3 major components: a `HttpClient`, a `AsyncRequestBuilder`, and a `ResponseHandler`.
+
+Gobblin offers 2 implementations of async
+http writers. As long as your write requirement can be expressed as a `HttpOperation` through a `Converter`, the
+2 implementations should work with configurations.
+
+## `AvroHttpWriterBuilder`
+An `AvroHttpWriterBuilder` builds an `AsyncHttpWriter` on top of the [apache httpcomponents framework](https://hc.apache.org/), sending vanilla http request.
+The 3 major components are:
+
+  - `ApacheHttpClient`. It uses [`CloseableHttpClient`](https://github.com/apache/httpcomponents-client/blob/master/httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/CloseableHttpClient.java) to 
+  send [`HttpUriRequest`](https://github.com/apache/httpcomponents-client/blob/master/httpclient5/src/main/java/org/apache/hc/client5/http/classic/methods/HttpUriRequest.java)
+  and receive [`CloseableHttpResponse`](https://github.com/apache/httpcomponents-client/blob/master/httpclient5/src/main/java/org/apache/hc/client5/http/impl/classic/CloseableHttpResponse.java)
+  - `ApacheHttpRequestBuilder`. It builds a `ApacheHttpRequest`, which is an `AsyncRequest` that wraps the `HttpUriRequest`, from one `HttpOperation`
+  - `ApacheHttpResponseHandler`. It handles a `HttpResponse`
+
+Configurations for the builder are:
+
+| Configuration | Description | Example
+|---|---|---|
+| `gobblin.writer.http.urlTemplate` | Required, the url template(schema and port included), together with `keys` and `queryParams`, to be resolved to request url | ```http://www.test.com/profiles/${memberId}``` |
+| `gobblin.writer.http.verb` | Required, the type of http request | get, update, delete, etc |
```

Comment:
is there a reference list of verb we can refer to?

Reply:
Links to verbs are added